src/main/java/seedu/address/model/appointment/UniqueAppointmentList.java

```diff
@@ -47,15 +56,25 @@ public void add(Appointment toAdd) {
         internalList.add(toAdd);
     }
 
-    // /**
-    //  * Replaces the appointment {@code target} in the list with {@code editedAppointment}.
-    //  * {@code target} must exist in the list.
-    //  * The appointment identity of {@code editedAppointment} must not be the same as
-    //  * another existing appointment in the list.
-    //  */
-    // public void setAppointment(Appointment target, Appointment editedAppointment) {
-    //     // todo
-    // }
+    /**
+     * Replaces the appointment {@code target} in the list with {@code editedAppointment}.
+     * {@code target} must exist in the list.
+     * The appointment identity of {@code editedAppointment} must not be the same as
+     * another existing appointment in the list.
+     */
+    public void setAppointment(Appointment target, Appointment editedAppointment) {
+        requireAllNonNull(target, editedAppointment);
+
+        int index = internalList.indexOf(target);
+        if (index == -1) {
+            throw new AppointmentNotFoundException();
+        }
+        if (target.equals(editedAppointment) && hasOverlaps(editedAppointment)) {
+            throw new OverlappingAppointmentException();
+        } else {
+            internalList.set(index, editedAppointment);
+        }
```

Comment:
The check should probably be `!target.startAtSameTime(editedAppointment) && hasOverlaps(editedAppointment)` instead. (If the edited version is different from the target, make sure that there is no overlap with the rest of the appointment.)

I think that the `else {` part can also be removed so that the happy path is made more prominent. 😄